providers/azure/azure.go

```diff
@@ -46,6 +46,8 @@ var DefaultConfig = Config{
 
 // Config Azure storage configuration.
 type Config struct {
+	ClientSecret            string             `yaml:"client_secret"`
```

Comment:
when authentication using `ClientSecret`, can we rely on a new `ClientID` field instead of reusing `UserAssignedID`. I think that makes the intent explicit

Reply:
good point. Pushed a commit adding it.